test/com/google/enterprise/adaptor/database/ResponseGeneratorTest.java

```diff
@@ -295,59 +426,66 @@ public void testUrlColumnModeServesResult() throws Exception {
+    URI testUri = testFile.toURI();
+    testUri = new URI(testUri.getScheme(), "localhost", testUri.getPath(),
+                      null);
 
-  @Test
-  public void testUrlAndMetadataListerModeServesResult() throws Exception {
-    executeUpdate("create table data(url varchar)");
-    executeUpdate("insert into data() values ()");
+    executeUpdate("create table data(url varchar, col1 varchar)");
+    String sql = "insert into data(url, col1) values (?, ?)";
+    try (PreparedStatement ps = getConnection().prepareStatement(sql)) {
+      ps.setString(1, testUri.toString());
+      ps.setString(2, "some metadata");
+      assertEquals(1, ps.executeUpdate());
+    }
 
     MockResponse uar = new MockResponse();
     Response response = newProxyInstance(Response.class, uar);
     Map<String, String> cfg = new TreeMap<String, String>();
+    cfg.put("db.metadataColumns", "col1");
     ResponseGenerator resgen = ResponseGenerator.urlAndMetadataLister(cfg);
     ResultSet rs = executeQueryAndNext("select * from data");
     resgen.generateResponse(rs, response);
     assertEquals(true, uar.notFound);
     assertEquals("", uar.baos.toString(UTF_8.name()));
-    assertEquals(null, uar.contentType);
-    assertEquals(null, uar.displayUrl);
+    assertNull(uar.contentType);
+    assertNull(uar.displayUrl);
   }
 
   @Test
-  public void testUrlAndMetadataListerModeColumnName() throws Exception {
+  public void testUrlAndMetadataLister_ignoresColumnName() throws Exception {
+    executeUpdate("create table data(url varchar)");
+    Map<String, String> cfg = new TreeMap<String, String>();
+    cfg.put("columnName", "anycolumn");
+    List<String> messages = new ArrayList<String>();
+    captureLogMessages(ResponseGenerator.class,
+        "urlAndMetadataLister mode ignores columnName", messages);
+    ResponseGenerator resgen = ResponseGenerator.urlAndMetadataLister(cfg);
+    assertEquals(messages.toString(), 1, messages.size());
+  }
+
+  @Test
+  public void testUrlAndMetadataLister_nullUrl() throws Exception {
     executeUpdate("create table data(url varchar)");
     executeUpdate("insert into data() values ()");
 
     MockResponse uar = new MockResponse();
     Response response = newProxyInstance(Response.class, uar);
     Map<String, String> cfg = new TreeMap<String, String>();
-    cfg.put("columnName", "anycolumn");
     ResponseGenerator resgen = ResponseGenerator.urlAndMetadataLister(cfg);
     ResultSet rs = executeQueryAndNext("select * from data");
     resgen.generateResponse(rs, response);
     assertEquals(true, uar.notFound);
     assertEquals("", uar.baos.toString(UTF_8.name()));
-    assertEquals(null, uar.contentType);
-    assertEquals(null, uar.displayUrl);
+    assertNull(uar.contentType);
+    assertNull(uar.displayUrl);
   }
 
   @Test
-  public void testBlobColumnModeContentTypeOverrideAndContentTypeCol()
-      throws Exception {
+  public void testContentTypeColumn_contentTypeOverride() throws Exception {
```

Comment:
The config property is "contentTypeCol", not "Column". These three tests are per-class specific, so I think these should actually retain the testContentColumn_ prefix, maybe testContentColumn_contentTypeConflict for this one? I'm happy to just add a TODO to add content type config tests for the others classes. (You did add a single contentTypeCol test for UrlColumn.)